edgraph/server.go

```diff
@@ -654,6 +654,21 @@ func doQueryInUpsert(ctx context.Context, req *api.Request, gmu *gql.Mutation) (
 		varToUID[name] = uids
 	}
 
+	varToVAL := make(map[string]map[string]string)
+	for name, v := range qr.Vars {
+		if v.Vals == nil || len(v.Vals) <= 0 {
+			continue
+		}
+
```

Comment:
please add a comment here as what we are storing in `varToVal`. An example would be useful too.